tcping.go

```diff
@@ -50,6 +99,10 @@ type stats struct {
 
 	// ticker is used to handle time between probes.
 	ticker *time.Ticker
+
+	// printer holds the choosen printer implementation for
+	// outputting information and data.
+	printer printer
```

Comment:
Just wondering, why we need to specify the type here?

This will result in using the printer like: `tcpStats.printer.printXYZ(...)`

Whereas by omitting the type, we can use it like: `tcpStats.printXYZ(...)`

Curious to know what are the impacts.

Reply:
Yeah, there's no difference except for what you mentioned, but there's a possibility of name collisions. We already have `printStats()` on `stats` and `printStatistics()` on `printer`, who knows what else will be implemented.

It's kind of a habit of mine to avoid situations like this, but I'm fine with removing the name. _not the type, they just happen to be called the same_

Sidenote: Such name collisions could still be resolved by calling methods using their type name explicitly, I just don't think it's clean code because it's inconsistent.

```go
type printer struct{}
func (p *printer) p() { fmt.Println(1) }

type stats struct{ printer }
func (s *stats) p() { fmt.Println(2) }

func main() {
	s := &stats{}
	s.printer.p() // 1
	s.p()         // 2
}
```

Reply:
Alright. Let's get this merged in.